### PR TITLE
Add int32 support and change format of castVarchar

### PR DESCRIPTION
**Pull request by rui-mo**
fixed https://github.com/oap-project/gazelle_plugin/issues/405.

### CODE REVIEW

**Comment by rui-mo** [comment]
TPC-DS q12 on double dataset:
Before fix:
Screenshot from 2021-07-12 17-13-29
After fix
Screenshot from 2021-07-12 17-14-32

File: cpp/src/gandiva/precompiled/arithmetic_ops.cc [hunk 1]
```diff
@@ -220,10 +220,10 @@ CAST_UNARY_UTF8(castVARCHAR, int8, utf8, "%d", NOFMT, nothing)
 CAST_UNARY_UTF8(castVARCHAR, int16, utf8, "%d", NOFMT, nothing)
 CAST_UNARY_UTF8(castVARCHAR, int32, utf8, "%d", NOFMT, nothing)
 CAST_UNARY_UTF8(castVARCHAR, int64, utf8, "%ld", NOFMT, nothing)
-// CAST_UNARY_UTF8(castVARCHAR, float32, utf8, "%.*f", FMT, FLT_DIG)
-// CAST_UNARY_UTF8(castVARCHAR, float64, utf8, "%.*f", FMT, DBL_DIG)
-CAST_UNARY_UTF8(castVARCHAR, float32, utf8, "%g", NOFMT, nothing)
-CAST_UNARY_UTF8(castVARCHAR, float64, utf8, "%g", NOFMT, nothing)
+CAST_UNARY_UTF8(castVARCHAR, float32, utf8, "%.*f", FMT, FLT_DIG)
+CAST_UNARY_UTF8(castVARCHAR, float64, utf8, "%.*f", FMT, DBL_DIG)
```
Comment (rui-mo) [inline cpp/src/gandiva/precompiled/arithmetic_ops.cc:224]: @zhouyuan Please take a look at this, thanks!

Comment (zhouyuan) [reply cpp/src/gandiva/precompiled/arithmetic_ops.cc:224]: can you please make a note here? this is required as Spark has some different behavior as gandiva

Comment (rui-mo) [reply cpp/src/gandiva/precompiled/arithmetic_ops.cc:226]: note was added

**Pull request merged**